zerver/lib/import_realm.py

```diff
@@ -444,6 +452,53 @@ def fix_message_rendered_content(
             )
 
 
+def valid_near_link_parts(near_link_parts: list[str]) -> bool:
+    return (
+        3 <= len(near_link_parts) <= 7
+        and near_link_parts[0] == "#narrow"
+        and near_link_parts[1] in ["channel", "stream"]
+    )
+
+
+def fix_near_links_in_messages(rendered_content: str) -> str:
+    soup = BeautifulSoup(rendered_content, "html.parser")
+    near_links = soup.find_all(
+        lambda tag: tag.name == "a"
+        and tag.has_attr("href")
+        and ("#narrow/stream/" in tag.get("href") or "#narrow/channel/" in tag.get("href"))
+    )
+
+    if near_links == []:
+        return rendered_content
+
+    message_id_map = ID_MAP["message"]
+    channel_id_map = ID_MAP["stream"]
+    for link in near_links:
+        href = link["href"]
+        # This is necessary because only some near-link start
+        # with "/".
+        near_link_parts = " ".join(href.split("/")).split()
```

Comment:
Hmm, I think all of them should start with `/`, see https://github.com/zulip/zulip/pull/32272/files#r1833247743

Reply:
form https://github.com/zulip/zulip/pull/32272#discussion_r1841689344:
>Ah, rewrite_local_links_to_relative includes the "/" as part of the prefix, so when removing it, the resulting relative link doesn't begin with "/".
> ...